Add mingw build to GitHub actions

On latest master, I have attempted a build on a clean install of Windows 10 using mingw installed via chocolately:

    mkdir build
    cd build
    cmake -G Ninja ..
    ninja -j8

Several warnings and the following error are emitted:
```
[7/213] Building CXX object check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj
FAILED: check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj
C:\ProgramData\chocolatey\bin\c++.exe  -IC:/Users/nmckibben/Documents/HiGHS/build -IC:/Users/nmckibben/Documents/HiGHS/app -IC:/Users/nmckibben/Documents/HiGHS/extern -IC:/Users/nmckibben/Documents/HiGHS/src -IC:/Users/nmckibben/Documents/HiGHS/src/io -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/ipx -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/basiclu -IC:/Users/nmckibben/Documents/HiGHS/src/lp_data -IC:/Users/nmckibben/Documents/HiGHS/src/mip -IC:/Users/nmckibben/Documents/HiGHS/src/model -IC:/Users/nmckibben/Documents/HiGHS/src/presolve -IC:/Users/nmckibben/Documents/HiGHS/src/qpsolver -IC:/Users/nmckibben/Documents/HiGHS/src/simplex -IC:/Users/nmckibben/Documents/HiGHS/src/test -IC:/Users/nmckibben/Documents/HiGHS/src/util -IC:/Users/nmckibben/Documents/HiGHS/src/extern/catch -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/ipx/src -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/basiclu/src -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/ipx/include -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/basiclu/include -Wall -Wextra -Wno-unused-parameter -Wno-format-truncation -pedantic -fdiagnostics-color=always -O3 -DNDEBUG -std=gnu++11 -MD -MT check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj -MF check\CMakeFiles\unit_tests.dir\TestIO.cpp.obj.d -o check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj -c C:/Users/nmckibben/Documents/HiGHS/check/TestIO.cpp
In file included from C:/Users/nmckibben/Documents/HiGHS/check/TestIO.cpp:5:
C:/Users/nmckibben/Documents/HiGHS/extern/catch.hpp: In instantiation of 'bool Catch::compareNotEqual(const LhsT&, RhsT&&) [with LhsT = char*; RhsT = const long long int&]':
C:/Users/nmckibben/Documents/HiGHS/extern/catch.hpp:2348:37:   required from 'const Catch::BinaryExpr<LhsT, const RhsT&> Catch::ExprLhs<LhsT>::operator!=(const RhsT&) [with RhsT = long long int; LhsT = char* const&]'
C:/Users/nmckibben/Documents/HiGHS/check/TestIO.cpp:66:5:   required from here
C:/Users/nmckibben/Documents/HiGHS/extern/catch.hpp:2322:96: error: ISO C++ forbids comparison between pointer and integer [-fpermissive]
```

Full output:
<details>

```
C:\Users\nmckibben\Documents\HiGHS\build>cmake -G Ninja ..
-- The CXX compiler identification is GNU 11.2.0
-- The C compiler identification is GNU 11.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/ProgramData/chocolatey/bin/c++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/ProgramData/chocolatey/bin/gcc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test HIGHS_HAVE_MM_PAUSE
-- Performing Test HIGHS_HAVE_MM_PAUSE - Success
-- Performing Test HIGHS_HAVE_BUILTIN_CLZ
-- Performing Test HIGHS_HAVE_BUILTIN_CLZ - Success
-- Performing Test flag_supported
-- Performing Test flag_supported - Success
-- Performing Test flag_supported
-- Performing Test flag_supported - Success
-- Performing Test flag_supported
-- Performing Test flag_supported - Success
-- Performing Test flag_supported
-- Performing Test flag_supported - Success
-- Performing Test flag_supported
-- Performing Test flag_supported - Success
-- Looking for a Fortran compiler
-- Looking for a Fortran compiler - C:/ProgramData/chocolatey/bin/gfortran.exe
-- The Fortran compiler identification is GNU 11.2.0
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: C:/ProgramData/chocolatey/bin/gfortran.exe - skipped
-- Looking for a CSharp compiler
-- Looking for a CSharp compiler - NOTFOUND
-- Performing Test NO_OMIT_FRAME_POINTER_FLAG_SUPPORTED
-- Performing Test NO_OMIT_FRAME_POINTER_FLAG_SUPPORTED - Success
-- Git hash: eeb455ff
-- Compilation date: 2021-12-10
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
-- Build shared libraries: ON
-- Build type: Release
-- No CSharp support
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/nmckibben/Documents/HiGHS/build

C:\Users\nmckibben\Documents\HiGHS\build>ninja -j4
[1/213] Building CXX object check/CMakeFiles/unit_tests.dir/TestEkk.cpp.obj
In file included from C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsLpSolverObject.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsRanging.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/Highs.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/check/TestEkk.cpp:1:
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h: In instantiation of 'class HighsHashTable<long long unsigned int>':
C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:139:28:   required from here
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:816:41: warning: 'this' pointer is null [-Wnonnull]
  816 |       reinterpret_cast<Entry*>(0)->value())>::type;
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:791:12: note: in a call to non-static member function 'const T& HighsHashTableEntry<T, void>::value() const [with T = long long unsigned int]'
  791 |   const T& value() const { return value_; }
      |            ^~~~~
[2/213] Building CXX object check/CMakeFiles/unit_tests.dir/TestDualise.cpp.obj
In file included from C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsLpSolverObject.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsRanging.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/Highs.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/check/TestDualise.cpp:1:
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h: In instantiation of 'class HighsHashTable<long long unsigned int>':
C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:139:28:   required from here
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:816:41: warning: 'this' pointer is null [-Wnonnull]
  816 |       reinterpret_cast<Entry*>(0)->value())>::type;
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:791:12: note: in a call to non-static member function 'const T& HighsHashTableEntry<T, void>::value() const [with T = long long unsigned int]'
  791 |   const T& value() const { return value_; }
      |            ^~~~~
[3/213] Building CXX object check/CMakeFiles/unit_tests.dir/TestAlienBasis.cpp.obj
In file included from C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsLpSolverObject.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsRanging.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/Highs.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/check/TestAlienBasis.cpp:3:
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h: In instantiation of 'class HighsHashTable<long long unsigned int>':
C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:139:28:   required from here
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:816:41: warning: 'this' pointer is null [-Wnonnull]
  816 |       reinterpret_cast<Entry*>(0)->value())>::type;
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:791:12: note: in a call to non-static member function 'const T& HighsHashTableEntry<T, void>::value() const [with T = long long unsigned int]'
  791 |   const T& value() const { return value_; }
      |            ^~~~~
[5/213] Building CXX object check/CMakeFiles/unit_tests.dir/TestCheckSolution.cpp.obj
In file included from C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsLpSolverObject.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsRanging.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/Highs.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/check/TestCheckSolution.cpp:3:
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h: In instantiation of 'class HighsHashTable<long long unsigned int>':
C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:139:28:   required from here
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:816:41: warning: 'this' pointer is null [-Wnonnull]
  816 |       reinterpret_cast<Entry*>(0)->value())>::type;
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:791:12: note: in a call to non-static member function 'const T& HighsHashTableEntry<T, void>::value() const [with T = long long unsigned int]'
  791 |   const T& value() const { return value_; }
      |            ^~~~~
[6/213] Building CXX object check/CMakeFiles/unit_tests.dir/TestFreezeBasis.cpp.obj
In file included from C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsLpSolverObject.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsRanging.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/Highs.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/check/TestFreezeBasis.cpp:1:
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h: In instantiation of 'class HighsHashTable<long long unsigned int>':
C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:139:28:   required from here
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:816:41: warning: 'this' pointer is null [-Wnonnull]
  816 |       reinterpret_cast<Entry*>(0)->value())>::type;
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:791:12: note: in a call to non-static member function 'const T& HighsHashTableEntry<T, void>::value() const [with T = long long unsigned int]'
  791 |   const T& value() const { return value_; }
      |            ^~~~~
[7/213] Building CXX object check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj
FAILED: check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj
C:\ProgramData\chocolatey\bin\c++.exe  -IC:/Users/nmckibben/Documents/HiGHS/build -IC:/Users/nmckibben/Documents/HiGHS/app -IC:/Users/nmckibben/Documents/HiGHS/extern -IC:/Users/nmckibben/Documents/HiGHS/src -IC:/Users/nmckibben/Documents/HiGHS/src/io -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/ipx -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/basiclu -IC:/Users/nmckibben/Documents/HiGHS/src/lp_data -IC:/Users/nmckibben/Documents/HiGHS/src/mip -IC:/Users/nmckibben/Documents/HiGHS/src/model -IC:/Users/nmckibben/Documents/HiGHS/src/presolve -IC:/Users/nmckibben/Documents/HiGHS/src/qpsolver -IC:/Users/nmckibben/Documents/HiGHS/src/simplex -IC:/Users/nmckibben/Documents/HiGHS/src/test -IC:/Users/nmckibben/Documents/HiGHS/src/util -IC:/Users/nmckibben/Documents/HiGHS/src/extern/catch -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/ipx/src -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/basiclu/src -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/ipx/include -IC:/Users/nmckibben/Documents/HiGHS/src/ipm/basiclu/include -Wall -Wextra -Wno-unused-parameter -Wno-format-truncation -pedantic -fdiagnostics-color=always -O3 -DNDEBUG -std=gnu++11 -MD -MT check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj -MF check\CMakeFiles\unit_tests.dir\TestIO.cpp.obj.d -o check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj -c C:/Users/nmckibben/Documents/HiGHS/check/TestIO.cpp
In file included from C:/Users/nmckibben/Documents/HiGHS/check/TestIO.cpp:5:
C:/Users/nmckibben/Documents/HiGHS/extern/catch.hpp: In instantiation of 'bool Catch::compareNotEqual(const LhsT&, RhsT&&) [with LhsT = char*; RhsT = const long long int&]':
C:/Users/nmckibben/Documents/HiGHS/extern/catch.hpp:2348:37:   required from 'const Catch::BinaryExpr<LhsT, const RhsT&> Catch::ExprLhs<LhsT>::operator!=(const RhsT&) [with RhsT = long long int; LhsT = char* const&]'
C:/Users/nmckibben/Documents/HiGHS/check/TestIO.cpp:66:5:   required from here
C:/Users/nmckibben/Documents/HiGHS/extern/catch.hpp:2322:96: error: ISO C++ forbids comparison between pointer and integer [-fpermissive]
 2322 |     auto compareNotEqual( LhsT const& lhs, RhsT&& rhs ) -> bool { return static_cast<bool>(lhs != rhs); }
      |                                                                                            ~~~~^~~~~~
[8/213] Building CXX object check/CMakeFiles/unit_tests.dir/TestHotStart.cpp.obj
In file included from C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsLpSolverObject.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsRanging.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/Highs.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/check/TestHotStart.cpp:1:
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h: In instantiation of 'class HighsHashTable<long long unsigned int>':
C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:139:28:   required from here
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:816:41: warning: 'this' pointer is null [-Wnonnull]
  816 |       reinterpret_cast<Entry*>(0)->value())>::type;
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:791:12: note: in a call to non-static member function 'const T& HighsHashTableEntry<T, void>::value() const [with T = long long unsigned int]'
  791 |   const T& value() const { return value_; }
      |            ^~~~~
[9/213] Building CXX object check/CMakeFiles/unit_tests.dir/TestOptions.cpp.obj
In file included from C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsLpSolverObject.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsRanging.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/Highs.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/check/TestOptions.cpp:4:
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h: In instantiation of 'class HighsHashTable<long long unsigned int>':
C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:139:28:   required from here
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:816:41: warning: 'this' pointer is null [-Wnonnull]
  816 |       reinterpret_cast<Entry*>(0)->value())>::type;
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:791:12: note: in a call to non-static member function 'const T& HighsHashTableEntry<T, void>::value() const [with T = long long unsigned int]'
  791 |   const T& value() const { return value_; }
      |            ^~~~~
[10/213] Building CXX object app/CMakeFiles/highs.dir/RunHighs.cpp.obj
In file included from C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsLpSolverObject.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsRanging.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/Highs.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/app/RunHighs.cpp:16:
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h: In instantiation of 'class HighsHashTable<long long unsigned int>':
C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:139:28:   required from here
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:816:41: warning: 'this' pointer is null [-Wnonnull]
  816 |       reinterpret_cast<Entry*>(0)->value())>::type;
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:791:12: note: in a call to non-static member function 'const T& HighsHashTableEntry<T, void>::value() const [with T = long long unsigned int]'
  791 |   const T& value() const { return value_; }
      |            ^~~~~
ninja: build stopped: subcommand failed.
```

</details>

Very similar errors when running explitly with MinGW Makefiles (which uses g++):

```
cmake -G "MinGW Makefiles" ..
mingw32-make.exe
```

Relevant error:

<details>

```
[ 79%] Linking CXX executable ..\bin\highs.exe
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x17c4): undefined reference to `highsLogUser(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x18c9): undefined reference to `HighsSparseMatrix::numNz() const'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x18d4): undefined reference to `HighsHessian::numNz() const'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x190b): undefined reference to `highsLogDev(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1923): undefined reference to `highsLogDev(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x193a): undefined reference to `highsLogDev(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x195a): undefined reference to `highsLogDev(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1989): undefined reference to `highsLogDev(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x19e0): undefined reference to `highsLogUser(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1a0c): undefined reference to `highsLogUser(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1a2c): undefined reference to `highsLogUser(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1a57): undefined reference to `highsLogUser(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1a6b): undefined reference to `highsLogUser(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1ac8): more undefined references to `highsLogUser(HighsLogOptions const&, HighsLogType, char const*, ...)' follow
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1b60): undefined reference to `highsLogDev(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1b77): undefined reference to `highsLogDev(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1b96): undefined reference to `highsLogDev(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1bb5): undefined reference to `highsLogDev(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x433c): undefined reference to `trim(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x441b): undefined reference to `setLocalOptionValue(HighsLogOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, HighsLogOptions&, std::vector<OptionRecord*, std::allocator<OptionRecord*> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x4512): undefined reference to `setLocalOptionValue(HighsLogOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, HighsLogOptions&, std::vector<OptionRecord*, std::allocator<OptionRecord*> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x45c4): undefined reference to `setLocalOptionValue(HighsLogOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<OptionRecord*, std::allocator<OptionRecord*> >&, double)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x4651): undefined reference to `loadOptionsFromFile(HighsLogOptions const&, HighsOptions&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x4713): undefined reference to `setLocalOptionValue(HighsLogOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, HighsLogOptions&, std::vector<OptionRecord*, std::allocator<OptionRecord*> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x490b): undefined reference to `setLocalOptionValue(HighsLogOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, HighsLogOptions&, std::vector<OptionRecord*, std::allocator<OptionRecord*> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x4c40): undefined reference to `setLocalOptionValue(HighsLogOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<OptionRecord*, std::allocator<OptionRecord*> >&, bool)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x4cfd): undefined reference to `setLocalOptionValue(HighsLogOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<OptionRecord*, std::allocator<OptionRecord*> >&, int)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x4db4): undefined reference to `setLocalOptionValue(HighsLogOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, HighsLogOptions&, std::vector<OptionRecord*, std::allocator<OptionRecord*> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x5016): undefined reference to `highsLogUser(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x17de): undefined reference to `highsLogUser(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text+0x1aad): undefined reference to `highsLogUser(HighsLogOptions const&, HighsLogType, char const*, ...)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text.startup+0x7b): undefined reference to `Highs::Highs()'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text.startup+0x18c): undefined reference to `HighsOptions::setLogOptions()'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text.startup+0x1f7): undefined reference to `Highs::passOptions(HighsOptions const&)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text.startup+0x231): undefined reference to `Highs::readModel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text.startup+0x265): undefined reference to `Highs::run()'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text.startup+0x311): undefined reference to `Highs::openLogFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text.startup+0x356): undefined reference to `Highs::getRanging()'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.text.startup+0x390): undefined reference to `Highs::writeSolution(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int)'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/11.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles\highs.dir/objects.a(RunHighs.cpp.obj):RunHighs.cpp:(.rdata$.refptr._ZTV17PresolveComponent[.refptr._ZTV17PresolveComponent]+0x0): undefined reference to `vtable for PresolveComponent'
collect2.exe: error: ld returned 1 exit status
mingw32-make[2]: *** [app\CMakeFiles\highs.dir\build.make:102: bin/highs.exe] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:173: app/CMakeFiles/highs.dir/all] Error 2
mingw32-make: *** [Makefile:165: all] Error 2
```

</details>

> Can you see if #706 fixed this?

@odow I just pulled latest HiGHS master and got the following error:
```
[ 80%] Building CXX object check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj
In file included from C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:22,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsLpSolverObject.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/lp_data/HighsRanging.h:21,
                 from C:/Users/nmckibben/Documents/HiGHS/src/Highs.h:22,
                 from C:\Users\nmckibben\Documents\HiGHS\check\TestIO.cpp:4:
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h: In instantiation of 'class HighsHashTable<long long unsigned int>':
C:/Users/nmckibben/Documents/HiGHS/src/simplex/HEkk.h:144:28:   required from here
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:816:41: warning: 'this' pointer is null [-Wnonnull]
  816 |       reinterpret_cast<Entry*>(0)->value())>::type;
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
C:/Users/nmckibben/Documents/HiGHS/src/util/HighsHash.h:791:12: note: in a call to non-static member function 'const T& HighsHashTableEntry<T, void>::value() const [with T = long long unsigned int]'
  791 |   const T& value() const { return value_; }
      |            ^~~~~
In file included from C:\Users\nmckibben\Documents\HiGHS\check\TestIO.cpp:5:
C:/Users/nmckibben/Documents/HiGHS/extern/catch.hpp: In instantiation of 'bool Catch::compareNotEqual(const LhsT&, RhsT&&) [with LhsT = char*; RhsT = const long long int&]':
C:/Users/nmckibben/Documents/HiGHS/extern/catch.hpp:2348:37:   required from 'const Catch::BinaryExpr<LhsT, const RhsT&> Catch::ExprLhs<LhsT>::operator!=(const RhsT&) [with RhsT = long long int; LhsT = char* const&]'
C:\Users\nmckibben\Documents\HiGHS\check\TestIO.cpp:89:5:   required from here
C:/Users/nmckibben/Documents/HiGHS/extern/catch.hpp:2322:96: error: ISO C++ forbids comparison between pointer and integer [-fpermissive]
 2322 |     auto compareNotEqual( LhsT const& lhs, RhsT&& rhs ) -> bool { return static_cast<bool>(lhs != rhs); }
      |                                                                                            ~~~~^~~~~~
mingw32-make[2]: *** [check\CMakeFiles\unit_tests.dir\build.make:211: check/CMakeFiles/unit_tests.dir/TestIO.cpp.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:929: check/CMakeFiles/unit_tests.dir/all] Error 2
mingw32-make: *** [Makefile:165: all] Error 2
```

--- a/.github/workflows/build-mingw.yml
+++ b/.github/workflows/build-mingw.yml
@@ -3,7 +3,7 @@ name: build-mingw
 on: ["push", "pull_request"]
 
 jobs:
-  mingw-idx32:
+  mingw:
     runs-on: windows-latest
 
     defaults:
@@ -18,7 +18,8 @@ jobs:
       matrix:
         msystem: [MINGW64, MINGW32, UCRT64, CLANG64]
         build-type: [Release, Debug]
-        fast-build: [ON, OFF]
+        fast-build: [On, Off]
+        int64: [On, Off]
         include:
           - msystem: MINGW64
             target-prefix: mingw-w64-x86_64
@@ -28,6 +29,11 @@ jobs:
             target-prefix: mingw-w64-ucrt-x86_64
           - msystem: CLANG64
             target-prefix: mingw-w64-clang-x86_64
+        exclude:
+          - msystem: MINGW32
+            int64: On
+          - fast-build: On
+            int64: On
 
     steps:
     - uses: msys2/setup-msys2@v2
@@ -42,77 +48,15 @@ jobs:
     - uses: actions/checkout@v3
 
     - name: Configure CMake
-      env:
-        BUILD_TYPE: ${{ matrix.build-type }}
-        FAST_BUILD: ${{ matrix.fast-build }}
       run: |
         mkdir build && cd build
-        cmake .. -DFAST_BUILD=${FAST_BUILD} -DCMAKE_BUILD_TYPE=${BUILD_TYPE}
+        cmake .. -DFAST_BUILD=${{ matrix.fast-build }} -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -DHIGHSINT64=${{ matrix.int64 }}
 
     - name: Build
-      env:
-        BUILD_TYPE: ${{ matrix.build-type }}
       # Execute the build.  You can specify a specific target with "--target <NAME>"
-      run: cd build && cmake --build . --config ${BUILD_TYPE} --parallel 4
+      run: cd build && cmake --build . --config ${{ matrix.build-type }} --parallel 4
 
     - name: Test
-      env:
-        BUILD_TYPE: ${{ matrix.build-type }}
       # Execute tests defined by the CMake configuration.
       # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
-      run: cd build && ctest --timeout 300 --output-on-failure -C ${BUILD_TYPE}
-
-  mingw-idx64:
-    runs-on: windows-latest
-
-    defaults:
-      run:
-        # Use MSYS2 as default shell
-        shell: msys2 {0}
-
-    strategy:
-      # Allow other runners in the matrix to continue if some fail
-      fail-fast: false
-
-      matrix:
-        msystem: [MINGW64, UCRT64, CLANG64]
-        build-type: [Release, Debug]
-        include:
-          - msystem: MINGW64
-            target-prefix: mingw-w64-x86_64
-          - msystem: UCRT64
-            target-prefix: mingw-w64-ucrt-x86_64
-          - msystem: CLANG64
-            target-prefix: mingw-w64-clang-x86_64
-
-    steps:
-    - uses: msys2/setup-msys2@v2
-      with:
-        msystem: ${{ matrix.msystem }}
-        install: >-
-          base-devel
-          ${{ matrix.target-prefix }}-cmake
-          ${{ matrix.target-prefix }}-cc
-          ${{ matrix.target-prefix }}-ninja
-
-    - uses: actions/checkout@v3
-
-    - name: Configure CMake
-      env:
-        BUILD_TYPE: ${{ matrix.build-type }}
-      run: |
-        mkdir build && cd build
-        cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DFAST_BUILD=OFF -DHIGHSINT64=on
-
-    - name: Build
-      env:
-        BUILD_TYPE: ${{ matrix.build-type }}
-      # Execute the build.  You can specify a specific target with "--target <NAME>"
-      run: cd build && cmake --build . --config ${BUILD_TYPE} --parallel 4
-
-    - name: Test
-      env:
-        BUILD_TYPE: ${{ matrix.build-type }}
-      # Execute tests defined by the CMake configuration.
-      # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
-      run: cd build && ctest --timeout 300 --output-on-failure -C ${BUILD_TYPE}
+      run: cd build && ctest --timeout 300 --output-on-failure -C ${{ matrix.build-type }}

--- a/.github/workflows/build-mingw.yml
+++ b/.github/workflows/build-mingw.yml
@@ -75,13 +75,11 @@ jobs:
       fail-fast: false
 
       matrix:
-        msystem: [MINGW64, MINGW32, UCRT64, CLANG64]
+        msystem: [MINGW64, UCRT64, CLANG64]
         build-type: [Release, Debug]
         include:
           - msystem: MINGW64
             target-prefix: mingw-w64-x86_64
-          - msystem: MINGW32
-            target-prefix: mingw-w64-i686
           - msystem: UCRT64
             target-prefix: mingw-w64-ucrt-x86_64
           - msystem: CLANG64

--- a/.github/workflows/build-mingw.yml
+++ b/.github/workflows/build-mingw.yml
@@ -16,22 +16,18 @@ jobs:
       fail-fast: false
 
       matrix:
-        msystem: [MINGW64, MINGW32, UCRT64, CLANG64]
+        msystem: [MINGW64, UCRT64, CLANG64]
         build-type: [Release, Debug]
         fast-build: [On, Off]
         int64: [On, Off]
         include:
           - msystem: MINGW64
             target-prefix: mingw-w64-x86_64
-          - msystem: MINGW32
-            target-prefix: mingw-w64-i686
           - msystem: UCRT64
             target-prefix: mingw-w64-ucrt-x86_64
           - msystem: CLANG64
             target-prefix: mingw-w64-clang-x86_64
         exclude:
-          - msystem: MINGW32
-            int64: On
           - fast-build: On
             int64: On
 

--- a/.github/workflows/build-mingw.yml
+++ b/.github/workflows/build-mingw.yml
@@ -1,0 +1,120 @@
+name: build-mingw
+
+on: ["push", "pull_request"]
+
+jobs:
+  mingw-idx32:
+    runs-on: windows-latest
+
+    defaults:
+      run:
+        # Use MSYS2 as default shell
+        shell: msys2 {0}
+
+    strategy:
+      # Allow other runners in the matrix to continue if some fail
+      fail-fast: false
+
+      matrix:
+        msystem: [MINGW64, MINGW32, UCRT64, CLANG64]
+        build-type: [Release, Debug]
+        fast-build: [ON, OFF]
+        include:
+          - msystem: MINGW64
+            target-prefix: mingw-w64-x86_64
+          - msystem: MINGW32
+            target-prefix: mingw-w64-i686
+          - msystem: UCRT64
+            target-prefix: mingw-w64-ucrt-x86_64
+          - msystem: CLANG64
+            target-prefix: mingw-w64-clang-x86_64
+
+    steps:
+    - uses: msys2/setup-msys2@v2
+      with:
+        msystem: ${{ matrix.msystem }}
+        install: >-
+          base-devel
+          ${{ matrix.target-prefix }}-cmake
+          ${{ matrix.target-prefix }}-cc
+          ${{ matrix.target-prefix }}-ninja
+
+    - uses: actions/checkout@v3
+
+    - name: Configure CMake
+      env:
+        BUILD_TYPE: ${{ matrix.build-type }}
+        FAST_BUILD: ${{ matrix.fast-build }}
+      run: |
+        mkdir build && cd build
+        cmake .. -DFAST_BUILD=${FAST_BUILD} -DCMAKE_BUILD_TYPE=${BUILD_TYPE}
+
+    - name: Build
+      env:
+        BUILD_TYPE: ${{ matrix.build-type }}
+      # Execute the build.  You can specify a specific target with "--target <NAME>"
+      run: cd build && cmake --build . --config ${BUILD_TYPE} --parallel 4
+
+    - name: Test
+      env:
+        BUILD_TYPE: ${{ matrix.build-type }}
+      # Execute tests defined by the CMake configuration.
+      # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
+      run: cd build && ctest --timeout 300 --output-on-failure -C ${BUILD_TYPE}
+
+  mingw-idx64:
+    runs-on: windows-latest
+
+    defaults:
+      run:
+        # Use MSYS2 as default shell
+        shell: msys2 {0}
+
+    strategy:
+      # Allow other runners in the matrix to continue if some fail
+      fail-fast: false
+
+      matrix:
+        msystem: [MINGW64, MINGW32, UCRT64, CLANG64]
+        build-type: [Release, Debug]
+        include:
+          - msystem: MINGW64
+            target-prefix: mingw-w64-x86_64
+          - msystem: MINGW32
+            target-prefix: mingw-w64-i686
+          - msystem: UCRT64
+            target-prefix: mingw-w64-ucrt-x86_64
+          - msystem: CLANG64
+            target-prefix: mingw-w64-clang-x86_64
+
+    steps:
+    - uses: msys2/setup-msys2@v2
+      with:
+        msystem: ${{ matrix.msystem }}
+        install: >-
+          base-devel
+          ${{ matrix.target-prefix }}-cmake
+          ${{ matrix.target-prefix }}-cc
+          ${{ matrix.target-prefix }}-ninja
+
+    - uses: actions/checkout@v3
+
+    - name: Configure CMake
+      env:
+        BUILD_TYPE: ${{ matrix.build-type }}
+      run: |
+        mkdir build && cd build
+        cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DFAST_BUILD=OFF -DHIGHSINT64=on
+
+    - name: Build
+      env:
+        BUILD_TYPE: ${{ matrix.build-type }}
+      # Execute the build.  You can specify a specific target with "--target <NAME>"
+      run: cd build && cmake --build . --config ${BUILD_TYPE} --parallel 4
+
+    - name: Test
+      env:
+        BUILD_TYPE: ${{ matrix.build-type }}
+      # Execute tests defined by the CMake configuration.
+      # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
+      run: cd build && ctest --timeout 300 --output-on-failure -C ${BUILD_TYPE}

